backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/inventory/InventoryRepository.java

```diff
@@ -742,4 +751,58 @@ public Set<InventoryId> getIdsByHUIdsAndDocTypeId(@NonNull final Set<HuId> huIds
 				.create()
 				.listIds(InventoryId::ofRepoId);
 	}
+
+	public Set<I_M_InventoryLine> retrieveAllLinesForHU(final @NonNull IContextAware ctxAware, @NonNull final HuId huId)
+	{
+		final List<I_M_HU> includedHUs = huDAO.retrieveIncludedHUs(huDAO.getById(huId));
+
+		final ImmutableSet<HuId> includedHUIds = includedHUs.stream()
+				.map(hu -> HuId.ofRepoId(hu.getM_HU_ID()))
+				.collect(ImmutableSet.toImmutableSet());
+
+		final ImmutableSet<HuId> huIds = ImmutableSet.<HuId>builder()
+				.addAll(includedHUIds)
+				.add(huId)
+				.build();
+
+		final ImmutableSet<InventoryLineId> linesViaHUId = queryBL.createQueryBuilder(I_M_InventoryLine.class)
+				.addInArrayFilter(I_M_InventoryLine.COLUMNNAME_M_HU_ID, huIds)
```

Comment:
please make sure that we have an index on each of
- `M_InventoryLine.M_HU_ID`
- `M_InventoryLine_HU.M_HU_ID`

Also i think there should be individual indices on each of
- `M_HU_Assignment.M_HU_ID`
- `M_HU_Assignment.M_LU_HU_ID`
- `M_HU_Assignment.M_TU_HU_ID`
- `M_HU_Assignment.VHU_ID`

(after a quick reseach, i hope that postgresql will be able to do something with them, particularly when the DB grew large)